watcher/watcher.go

```diff
@@ -1247,32 +1223,20 @@ func (w *Watcher) EnsureCorrectRooms() error {
 		return err
 	}
 
-	// get invalid pods (wrong versions and pods not registered in redis)
-	logger.Info("searching for invalid pods")
-	incorrectPods, unregisteredPods, err := w.getIncorrectAndUnregisteredPods(logger, pods, scheduler)
+	// get invalid pods (wrong versions)
+	incorrectPods, err := w.podsOfIncorrectVersion(pods, scheduler)
 	if err != nil {
 		logger.WithError(err).Error("failed to get invalid pods")
 		return err
 	}
 
-	// get incorrect pod names
-	var incorrectPodNames []string
-	if len(incorrectPods) > 0 {
-		for _, pod := range incorrectPods {
-			incorrectPodNames = append(incorrectPodNames, pod.Name)
-		}
-	}
-
-	// get unregistered pod names
-	var unregisteredPodNames []string
-	if len(unregisteredPods) > 0 {
-		for _, pod := range unregisteredPods {
-			unregisteredPodNames = append(unregisteredPodNames, pod.Name)
-		}
+	if len(incorrectPods) <= 0 {
```

Comment:
`len()` doesn't return negative numbers. You can replace it by:

```suggestion
	if len(incorrectPods) == 0 {
```